kong/api/routes/kong.lua

```diff
@@ -56,6 +57,7 @@ return {
         tagline = tagline,
         version = version,
         hostname = utils.get_hostname(),
+        node_id = public.get_node_id(),
```

Comment:
Should we handle the error here? Granted it is unlikely that `get_node_id()` will never be invoked before this particular call, but there is no safeguard against it if things ever change...

Reply:
I thought about that, but if we can't get a `node_id` here there's not much we can do other than just not returning the field, so returning nil naturally "erases" that field from the result (the only other thing that comes to mind would be to log the error; we could do that, but it seems overkill, and we don't for `utils.get_hostname()` anyway).

With the current design of `get_node_id`, even if this was the first time it is called, it wouldn't be a problem, as the node_id would be generated and memoized (assuming of course that route code is not invoked from the init phase :) )

Reply:
True, but when raising this concern, I am mostly worried about writing reference code - another typical pattern we have seen in this project has been the copy-pasting of countless bits of Lua code around the project - spreading bad patterns all around the place...

Reply:
Knowing when to handle an error and when not to is not a bad pattern per se, but it takes experience -- so I see what you mean. Added the check here, no biggie.

> another typical pattern we have seen in this project has been the copy-pasting of countless bits of Lua code around the project - spreading bad patterns all around the place...

Yeah, and ultimately, we should strive to reduce the copy-pasting rather than just providing people better bits to replicate. :)

Reply:
> Yeah, and ultimately, we should strive to reduce the copy-pasting rather than just providing people better bits to replicate. :)

My struggle for the past 3 years :) Never did we get allocated time to actually work on this - glad we will soon